tests/Integration/Behaviour/Features/Context/Domain/OrderPaymentFeatureContext.php

```diff
@@ -0,0 +1,214 @@
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+class OrderPaymentFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add payment to order with id :orderId with the following properties:
+     *
+     * @param int $orderId
+     * @param TableNode $table
+     *
+     * @throws RuntimeException
+     * @throws ServiceCircularReferenceException
+     * @throws ServiceNotFoundException
+     */
+    public function iAddPaymentToWithIdOrderWithIdTheFollowingProperties(int $orderId, TableNode $table)
+    {
+        /** @var array $hash */
+        $hash = $table->getHash();
+        if (count($hash) != 1) {
+            throw new RuntimeException('Payment details are invalid');
+        }
+        /** @var array $data */
+        $data = $hash[0];
+
+        $this->getCommandBus()->handle(
+            new AddPaymentCommand(
+                $orderId,
+                $data['date'],
+                $data['payment_method'],
+                $data['amount'],
+                (int) $data['id_currency'],
+                $data['id_invoice'],
+                $data['transaction_id']
+            )
+        );
+    }
+
+    /**
+     * @Then if I query order with id :orderId payments I should get :numberOfPayments payments
+     *
+     * @param int $orderId
+     * @param int $numberOfPayments
+     *
+     * @throws RuntimeException
+     * @throws ServiceCircularReferenceException
+     * @throws ServiceNotFoundException
+     */
+    public function ifIQueryOrderWithIdPaymentsIShouldGetPayments(int $orderId, int $numberOfPayments)
+    {
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+        /** @var OrderPaymentsForViewing $orderPaymentsForViewing */
+        $orderPaymentsForViewing = $orderForViewing->getPayments();
+        /** @var OrderPaymentForViewing[] $orderPaymentForViewingArray */
+        $orderPaymentForViewingArray = $orderPaymentsForViewing->getPayments();
+
+        $countOfOrderPaymentsFromDb = count($orderPaymentForViewingArray);
+        if (count($orderPaymentForViewingArray) !== $numberOfPayments) {
+            throw new RuntimeException(sprintf(
+                'Order "%s" number of payments  is "%s", but "%s" was expected',
+                $orderId,
+                $countOfOrderPaymentsFromDb,
+                $numberOfPayments
+            ));
+        }
+    }
+
+    /**
+     * @Then if I query order with id :orderId payments I should get an Order with properties:
+     *
+     * @param int $orderId
+     * @param TableNode $table
+     *
+     * @throws RuntimeException
+     * @throws ServiceCircularReferenceException
+     * @throws ServiceNotFoundException
+     */
+    public function ifIQueryOrderWithIdPaymentsIShouldGetAnOrderWithProperties(int $orderId, TableNode $table)
```

Comment:
Can you try to use `DataComparator` functions, like in https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php#L197 ?

And if it does not do the job ... do you think you can try to improve it ? The kind of complex comparisons you are doing here, we can guess we'll have to do many more, so let's try to have something reusable to do it for us 😄

Reply:
For start will create OrderPaymentForViewing object from table properties and use PHPUnit comparator (assertEquals) to compare those two objects. Let's see the power of PHPUnit object comparator :) The creation of object will be reusable.

That's the quickest solution. Might not though the best.